### DISCUSSION TITLE
Behavior of Global Auto-Type shortcut

### DISCUSSION THREAD

**Discussion by LMurphy001** [keepassxreboot/keepassxc#8414] [Q&A]
I'm not sure if this is a bug, or if I need to fix my settings.
If it's a bug, please indicate where/how to report it. If it's isn't a bug, I need to learn more on the subject of invoking auto-type using Global Auto-Type shortcut. Setting it up, invoking it, what it does when I invoke it, how entires are chosen for its list, how I should interact with it when the list appears)

My scenario:
Settings:

1. I defined my Global Auto-Type shortcut Ctrl-Alt-Shift-E
2. I unchecked "Use entry titles to match ...."
3. I clicked OK
 

1. I found an entry (Grammarly) and double-clicked its url. The URL is unique.
2. In Firefox, I clicked on the username box.
3. I pressed Ctrl-Alt-Shift-E
4. A list appeared. MANY MANY entries are in the list! I use the "Search" in the list to find the needle in the haystack
5. I selected Grammarly in the list
6. I pressed Ctrl-1 and the usernane is entered
7. The grammarly password box appeared, I clicked into it
8. I pressed Ctrl-Alt-Shift-E
9. The list appeared again. Grammary was still selected.
10. I pressed Ctrl-2 and the password is entered

My expectation was that the global shortcut Ctrl-Alt-Shift-E would only match by URL.

I'm assuming keepassxc already has that feature but that I don't have keepassxc correctly configured. How do I correctly configure it?

If it has the feature but I've encountered a bug, how and where do I report this?

If it doesn't have that feature, I would like to request it. How and where do I do so? (If it has already been discussed and voted down, please provide link so I can understand rationale)

Thank you

**Comment by droidmonkey** [comment]
Please read the user guide, everything is explained there. Global Auto-Type matches by _window title_. If the URL is NOT in the window title then it won't match. We recommend just defining window associations or you can invoke global Auto-Type from the browser extension itself. You can define a shortcut for Auto-Type from the extension or use the right click menu.

**Comment by LMurphy001** [comment]
Thank you. I found this described in the User Guide at: https://keepassxc.org/docs/KeePassXC_UserGuide.html#_configure_auto_type_sequences
It describes defining window title associations and optionally adding custom auto-type sequences for each. It starts here with "To configure Auto-Type sequences for your entries, perform the following steps:"

Turns out that Auto-type doesn't do what I thought it did. It doesn't match on URL unless I install a browser extension (add-on) which puts the URL into the browser window title bar.

I need to discover which of the non-ideal alternatives is liked least and rule it out.
* Auto-type with its inexact matching on title... which provides no guarantee the browser window is really what it says it is.
* Auto-type with better matching on title as long as I install an extension which stuffs at least the host name from the URL into the browser's window title. More secure but somewhat less private.
* Use the KeepassXC-Browser Add-On. This is non-ideal because it wants clipboard permissions, and is a lot more coding and complexity than, for example, https://github.com/Aaron-P/HostnameTitleBar

I have to weigh not only my own preferences, but those of others I'll be supporting...
